src/js/controller/controller.js

```diff
@@ -955,6 +955,14 @@ Object.assign(Controller.prototype, {
         this.getWidth = () => _model.get('containerWidth');
         this.getHeight = () => _model.get('containerHeight');
         this.getItemQoe = () => _model._qoeItem;
+
+        this.setItemCallback = function (callback) {
+            _programController.itemCallback = callback;
+        };
+        this.getItemPromise = function (index = (_model.get('item') + (_model.get('itemReady') ? 1 : 0))) {
```

Comment:
this is a style thing so no need to change but i think it's a bit tough to parse having the default be so complex but inlined here. just a thought!

Reply:
We could just require an index. What do you think?

Reply:
i think i'd prefer that as long as that work doesn't grow in complexity being outside the scope of the parent function

Reply:
Will make this require `index`

Reply:
@jnatalzia I made the change, and made sure to also support `getItemPromise(-1)` for the `"nextUp"` item. I'm a little worried that if someone uses this API they won't want to thing about if they are in a playlist or recs, but we can always make index optional and work out those details.